File: src/main/java/com/zakrywilson/systemctl/service/generator/SystemCtlService.java
package com.zakrywilson.systemctl.service.generator;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * @author Zach Wilson
 */
public class SystemCtlService {

    private long id;
    private String serviceFileContent;

    public SystemCtlService() {}

    public SystemCtlService(long id, String serviceFileContent) {
        this.id = id;
        this.serviceFileContent = serviceFileContent;
    }

    @JsonProperty
    public long getId() {
        return id;
    }

    @JsonProperty
    public String getServiceFileContent() {
        return serviceFileContent;
    }

}

File: src/main/kotlin/com/zakrywilson/systemctl/service/generator/models/SystemCtlService.kt
package com.zakrywilson.systemctl.service.generator.models

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Represents a single SystemCtl service.
 *
 * @author Zach Wilson
 */
data class SystemCtlService(@JsonProperty val id: Long,
                            @JsonProperty val name: String,
                            @JsonProperty val content: String,
                            @JsonProperty val jarFileName: String)
